src/main/java/net/dries007/tfc/util/interaction/InteractionManager.java

```diff
@@ -216,8 +233,11 @@ public static void onRightClickItem(PlayerInteractEvent.RightClickItem event)
             {
                 result = ServerInteractionManager.processRightClickItem(event, action);
             }
-            event.setCancellationResult(result);
-            event.setCanceled(true);
+            if (result != EnumActionResult.PASS)
+            {
+                event.setCancellationResult(result);
+                event.setCanceled(true);
+            }
```

Comment:
This is intentional. Our right click handling exactly simulates the call / execution path for vanilla's right click, so we can replace the call to `Item#onRightClick` at the appropriate time. (see 93502c5c0cf070c0d874793d891d1afc313c4ac8). As such, we need to cancel the event afterwards, otherwise it would potentially call to the item / block twice. In addition, we register this event handler at the lowest priority, so if other mods intercept the event they will take precedence and not trigger our special logic.